tests/unit/components/sl-menu-test.js

```diff
@@ -1,263 +1,565 @@
+    });
+
+    assert.equal(
+        component.get( 'selectedItem' ),
+        testItems[ 0 ].items[ 0 ],
+        'Selected item is first sub-menu item'
     );
 });
 
-test( 'Menu properly hides all but root list', function( assert ) {
-    this.subject({ menu: modelStub });
+test( 'selectLeft() selects an item in the "left" direction', function( assert ) {
+    const component = this.subject({
+        items: testItems
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        0,
+        'Nothing is selected initially'
+    );
+
+    Ember.run( () => {
+        // Selection has to be "down" first in order to select the top level for
+        // the context
+        component.selectDown();
+        component.selectLeft();
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        1,
+        'Only one selection is made'
+    );
+
+    Ember.run( () => {
+        // Select "left" to the first top-level item
+        component.selectLeft();
+
+        // Select "down" into the first sub-menu
+        component.selectDown();
+    });
+
+    // This assertion ensures that we have gone into the first sub-menu
+    assert.equal(
+        component.get( 'selections' ).length,
+        2,
+        'Two selections are made'
+    );
+
+    Ember.run( () => {
+        // Back to top-level items
+        component.selectLeft();
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        1,
+        'Only one selection is made'
+    );
 
     assert.equal(
-        this.$( 'li:visible' ).length,
-        3
+        component.get( 'selectedItem' ),
+        testItems[ 0 ],
+        'First top-level item is selected again'
     );
 });
 
-test( 'Menu shows child on hover', function( assert ) {
-    this.subject({ menu: modelStub });
+test( 'selectNext() selects the next sibling menu item', function( assert ) {
+    const component = this.subject({
+        items: testItems
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        0,
+        'Nothing is selected initially'
+    );
+
+    Ember.run( () => {
+        component.selectNext();
+    });
 
     assert.equal(
-        this.$( 'li:visible' ).length,
-        3
+        component.get( 'selectedItem' ),
+        testItems[ 0 ],
+        'First top-level item is selected'
     );
 
-    this.$( 'li:visible' ).first().mouseenter();
+    Ember.run( () => {
+        component.selectNext();
+    });
+
+    assert.equal(
+        component.get( 'selections' ).length,
+        1,
+        'One item is still selected'
+    );
 
     assert.equal(
-        this.$( 'li:visible' ).length,
-        7
+        component.get( 'selectedItem' ),
+        testItems[ 1 ],
+        'Second top-level item is selected'
     );
 });
 
-test( 'Menu closes child on mouse exit', function( assert ) {
-    this.subject({ menu: modelStub });
+test( "selectParent() selects a sub-menu's parent item", function( assert ) {
```

Comment:
No, it shouldn't. Note the apostrophe in "sub-menu's". ESLint's [quotes](http://eslint.org/docs/rules/quotes) rule specifically allows this.